prow/cluster/prow_job.yaml

```diff
@@ -10,3 +10,31 @@ spec:
     singular: prowjob
     plural: prowjobs
   scope: Namespaced
+  validation:
+      openAPIV3Schema:
+        properties:
+          spec:
+            properties:
+              max_concurrency:
+                type: integer
+                minimum: 0
+              type:
+                type: string
+                pattern: '^(presubmit|postsubmit|periodic|batch)$'
+          status:
+            properties:
+              state:
+                type: string
+                pattern: '^(triggered|pending|success|failure|aborted|error)$'
+            anyOf:
+            - not:
+                properties:
+                  state:
+                    type: string
+                    enum:
+                    - "success"
+                    - "failure"
+                    - "error"
+                    - "aborted"
+            - required:
+              - completionTime
```

Comment:
Does this mean that this is required all the time? Running prowjobs don't have a completion time, but it's probably set to the zero value somewhere.

Reply:
I think the `anyOf` operator handles this. Either `completionTime` must be specified or `state` must not be "success", "failure", "error", or "aborted".

Reply:
> I think the anyOf operator handles this. Either completionTime must be specified or state must not be "success", "failure", "error", or "aborted".

Yes, this is what happens.

Reply:
@spxtr are you worried that running prowjobs may be invalidated? It should be easy to verify it and we should probably stop setting it to the zero value.

Reply:
Hrm, the issue is probably with Go defaulting. Any arguments against turning [CompletionTime](https://github.com/kubernetes/test-infra/blob/f72668786f9002196e04337728b3c7a19ecacdd3/prow/kube/prowjob.go#L96) into a pointer?

Reply:
Opened https://github.com/kubernetes/test-infra/pull/6215

Reply:
Why is there a problem with this? Shouldn't the `omitempty` annotation prevent the `CompletionTime` field from being included if it is 0?

Reply:
Go structs do not work with `omitempty` as expected: https://stackoverflow.com/questions/32643815/golang-json-omitempty-with-time-time-field/32646035#32646035

Reply:
Also, you can verify this today by doing a `kubectl get prowjob <name> -o yaml` on a pending/triggered prowjob. `CompletionTime` is set (ie. not dropped by `omitempty`).